neuralop/layers/segment_csr.py

```diff
@@ -54,38 +60,23 @@ def segment_csr(
         # if batched, shape [b, n_reps, channels]
         # otherwise shape [n_reps, channels]
         if src.ndim == 3:
-            batched = True
             point_dim = 1
         else:
-            batched = False
             point_dim = 0
 
         # if batched, shape [b, n_out, channels]
         # otherwise shape [n_out, channels]
         output_shape = list(src.shape)
-        n_out = indptr.shape[point_dim] - 1
+        n_out = indptr.shape[0] - 1
+        n_reps = indptr[1:] - indptr[:-1]
```

Comment:
Shall we use a more informative name for indptr and add some more comments on what we do?